src/validateOptions.js

```diff
@@ -22,17 +19,29 @@ const ajv = new Ajv({
 errors(ajv);
 keywords(ajv, ['instanceof', 'typeof']);
 
-const validateOptions = (schema, options, name) => {
-  if (typeof schema === 'string') {
-    schema = fs.readFileSync(path.resolve(schema), 'utf8');
-    schema = JSON.parse(schema);
-  }
+const passedBySchema = new WeakMap();
 
-  if (!ajv.validate(schema, options)) {
-    throw new ValidationError(ajv.errors, name);
+const validateOptions = (schema, options, name) => {
+  const passed = passedBySchema.get(schema);
+
+  if (passed) {
+    if (!passed.has(options)) {
+      if (!ajv.validate(schema, options)) {
+        throw new ValidationError(ajv.errors, name);
+      }
+
+      passed.add(options);
+    }
+  } else {
+    if (!ajv.validate(schema, options)) {
+      throw new ValidationError(ajv.errors, name);
+    }
+
+    passedBySchema.set(schema, new WeakSet([options]));
   }
 
   return true;
 };
 
 module.exports = validateOptions;
+module.exports.ajv = ajv;
```

Comment:
Why exposing ajv?

Reply:
@sokra so it can be spied: https://github.com/webpack-contrib/schema-utils/blob/583f633d558d2e57cf272b175a3897c31fc1014d/test/index.test.js#L29

Reply:
@hulkish it is possible spy without export?